crates/services/txpool/src/txpool.rs

```diff
@@ -354,6 +369,21 @@ where
             return Err(Error::NotInsertedTxKnown)
         }
 
+        // If upgrading transition function, at least check that it is valid wasm
+        if let PoolTransaction::Upgrade(upgrade, _) = tx.as_ref() {
```

Comment:
We should include a test:
```
insert_inner__rejects_upgrade_tx_with_invalid_wasm()
```

Should be super simple if you change the `MockWasmChecker` to hold a `Result` and return that `Result`.

Reply:
Added in 030ceaee1